chemprop/cli/train.py

```diff
@@ -906,6 +912,10 @@ def train_model(
         save_model(p_model, model)
         logger.info(f"Best model saved to '{p_model}'")
 
+        if args.no_checkpoint:
+            logger.debug(f"Remove '{checkpointing.dirpath}' folder")
+            shutil.rmtree(checkpointing.dirpath)
```

Comment:
I don't think this is a very elegant solution to the problem. That is, it seems like you actually _do_ want checkpointing during training but just that you want them _deleted_ afterwards? If so, I think the flag should be named `--remove-checkpoints`. Or perhaps more broadly this can be handled via a `$TMPDIR` location and doesn't necessitate a CLI option

Reply:
Yes, this PR is intended to remove the checkpoints afterward. Using `$TMPDIR` is a good suggestion, and I can add code like
```python
dirpath = os.environ.get("TMPDIR", model_output_dir)
checkpoint_dir = os.path.join(dirpath, "checkpoints")
checkpointing = ModelCheckpoint(
    checkpoint_dir ,
    "best-{epoch}-{val_loss:.2f}",
    "val_loss",
    mode=monitor_mode,
    save_last=True,
)
```

Reply:
I think you're looking for the [`tempfile`](https://docs.python.org/3/library/tempfile.html) module.